src/getFieldsFromData.js

```diff
@@ -0,0 +1,41 @@
+import getTypeFromValue from './getTypeFromValue';
+
+/**
+ * Get a list of GraphQL fields from a list of entities
+ * 
+ * @example
+ * const entities = [
+ *     {
+ *         "id": 1,
+ *         "title": "Lorem Ipsum",
+ *         "views": 254,
+ *         "user_id": 123,
+ *     },
+ *     {
+ *         "id": 2,
+ *         "title": "Sic Dolor amet",
+ *         "views": 65,
+ *         "user_id": 456,
+ *     },
+ * ];
+ * const types = getFieldsFromData(entities);
+ * // [
+ *     id: { type: graphql.GraphQLString },
+ *     title: { type: graphql.GraphQLString },
+ *     views: { type: graphql.GraphQLInt }
+ *     user_id: { type: graphql.GraphQLString }
+ * ];
+ */
+export default data => {
+    const fields = {};
+    data.map(record => {
```

Comment:
why mapping over the data items ? Are you planning to retrieve a single record definition by checking for each field the type inferred most often ?